tests/integration/golang/aim/namespace/flows/app_test.go

```diff
@@ -0,0 +1,280 @@
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&resp,
+		).DoRequest(
+			"/apps",
+		),
+	)
+	// only app 2 should be present
+	assert.Equal(s.T(), 1, len(resp))
+	assert.Equal(s.T(), app2ID, resp[0].ID)
+
+	// IDs from other namespace cannot be fetched, updated, or deleted
+	errResp := response.Error{}
+	client := s.AIMClient()
+	require.Nil(
+		s.T(),
+		client.WithMethod(
+			http.MethodGet,
+		).WithNamespace(
+			namespace1Code,
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&errResp,
+		).DoRequest(
+			fmt.Sprintf("/apps/%s", app2ID),
+		),
+	)
+	assert.Equal(s.T(), fiber.ErrNotFound.Code, client.GetStatusCode())
+
+	client = s.AIMClient()
+	require.Nil(
+		s.T(),
+		client.WithMethod(
+			http.MethodPut,
+		).WithNamespace(
+			namespace2Code,
+		).WithRequest(
+			request.UpdateApp{
+				Type: "app-type",
+				State: request.AppState{
+					"app-state-key": "new-app-state-value",
+				},
+			},
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&errResp,
+		).DoRequest(
+			fmt.Sprintf("/apps/%s", app1ID),
+		),
+	)
+	assert.Equal(s.T(), fiber.ErrNotFound.Code, client.GetStatusCode())
+
+	client = s.AIMClient()
+	require.Nil(
+		s.T(),
+		client.WithMethod(
+			http.MethodDelete,
+		).WithNamespace(
+			namespace2Code,
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&errResp,
+		).DoRequest(
+			fmt.Sprintf("/apps/%s", app1ID),
+		),
+	)
+	assert.Equal(s.T(), fiber.ErrNotFound.Code, client.GetStatusCode())
+
+	// IDs from active namespace can be fetched, updated, and deleted
+	appResp := response.App{}
+	client = s.AIMClient()
+	require.Nil(
+		s.T(),
+		client.WithMethod(
+			http.MethodGet,
+		).WithNamespace(
+			namespace1Code,
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&appResp,
+		).DoRequest(
+			fmt.Sprintf("/apps/%s", app1ID),
+		),
+	)
+	assert.Equal(s.T(), app1ID, appResp.ID)
+	assert.Equal(s.T(), fiber.StatusOK, client.GetStatusCode())
+
+	client = s.AIMClient()
+	require.Nil(
+		s.T(),
+		client.WithMethod(
+			http.MethodPut,
+		).WithNamespace(
+			namespace1Code,
+		).WithRequest(
+			request.UpdateApp{
+				Type: "app-type",
+				State: request.AppState{
+					"app-state-key": "new-app-state-value",
+				},
+			},
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&appResp,
+		).DoRequest(
+			fmt.Sprintf("/apps/%s", app1ID),
+		),
+	)
+	assert.Equal(s.T(), app1ID, appResp.ID)
+	assert.Equal(s.T(), fiber.StatusOK, client.GetStatusCode())
+
+	client = s.AIMClient()
+	require.Nil(
+		s.T(),
+		client.WithMethod(
+			http.MethodDelete,
+		).WithNamespace(
+			namespace2Code,
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&appResp,
+		).DoRequest(
+			fmt.Sprintf("/apps/%s", app2ID),
+		),
+	)
+	assert.Equal(s.T(), fiber.StatusOK, client.GetStatusCode())
+}
+
+func (s *AppFlowTestSuite) createApp(namespace string, req *request.CreateApp) string {
```

Comment:
please rename methods where you compare to have `AndCompare` in the name so we can understand what is going on.

Reply:
done